### ISSUE TITLE
Throw an error when Fluent patterns are used outside of the a11y key

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/chipper#1613]
Fluent patterns should not be translatable yet and are only available for accessibility content. We should make sure that fluent patterns are not used outside of the a11y key. It should be possible to throw an error during modulify if a fluent pattern is used outside of the a11y key.

**Comment by samreid** [comment]
I tested `stopwatch:                             Stopwatch { $hello }` in forces-and-motion-basics-yaml outside of the a11y key, and it worked fine. This is a legal string.

However `{ myString }` is valid fluent syntax and valid json syntax (even though we typically have a number like `{ 0 }`, so we cannot error on that.

So perhaps we should just guard against `{ $string` patterns? or maybe `{ test }` where test in non-numeric?

**Comment by samreid** [comment]
This was done as part of https://github.com/phetsims/chipper/issues/1632
